discovery/manager_test.go

```diff
@@ -762,51 +760,110 @@ func TestTargetSetRecreatesTargetGroupsEveryRun(t *testing.T) {
 	discoveryManager.updatert = 100 * time.Millisecond
 	go discoveryManager.Run()
 
-	c := map[string]sd_config.ServiceDiscoveryConfig{
+	c := map[string]discoverer.ServiceDiscoveryConfig{
 		"prometheus": {
-			StaticConfigs: []*targetgroup.Group{
-				{
-					Source: "0",
-					Targets: []model.LabelSet{
+			Configs: []discoverer.Config{
+				&discoverer.StaticConfig{
+					TargetGroups: []*targetgroup.Group{
 						{
-							model.AddressLabel: model.LabelValue("foo:9090"),
+							Source: "0",
+							Targets: []model.LabelSet{
+								{
+									model.AddressLabel: model.LabelValue("foo:9090"),
+								},
+							},
+						},
+						{
+							Source: "1",
+							Targets: []model.LabelSet{
+								{
+									model.AddressLabel: model.LabelValue("bar:9090"),
+								},
+							},
 						},
 					},
 				},
-				{
-					Source: "1",
+			},
+		},
+	}
+	discoveryManager.ApplyConfig(c)
+
+	<-discoveryManager.SyncCh()
+	verifyPresence(t, discoveryManager.targets, poolKey{setName: "prometheus", provider: "static/0"}, "{__address__=\"foo:9090\"}", true)
+	verifyPresence(t, discoveryManager.targets, poolKey{setName: "prometheus", provider: "static/0"}, "{__address__=\"bar:9090\"}", true)
+
+	c["prometheus"] = discoverer.ServiceDiscoveryConfig{
+		Configs: []discoverer.Config{
+			&discoverer.StaticConfig{
+				TargetGroups: []*targetgroup.Group{{
+					Source: "0",
 					Targets: []model.LabelSet{
 						{
-							model.AddressLabel: model.LabelValue("bar:9090"),
+							model.AddressLabel: model.LabelValue("foo:9090"),
 						},
 					},
-				},
+				}},
 			},
 		},
 	}
 	discoveryManager.ApplyConfig(c)
 
 	<-discoveryManager.SyncCh()
-	verifyPresence(t, discoveryManager.targets, poolKey{setName: "prometheus", provider: "string/0"}, "{__address__=\"foo:9090\"}", true)
-	verifyPresence(t, discoveryManager.targets, poolKey{setName: "prometheus", provider: "string/0"}, "{__address__=\"bar:9090\"}", true)
+	verifyPresence(t, discoveryManager.targets, poolKey{setName: "prometheus", provider: "static/0"}, "{__address__=\"foo:9090\"}", true)
+	verifyPresence(t, discoveryManager.targets, poolKey{setName: "prometheus", provider: "static/0"}, "{__address__=\"bar:9090\"}", false)
+}
+
+type staticConfig []*targetgroup.Group
```

Comment:
Should we move the static sd to a new file?

Reply:
These tests should've been using `discoverer.StaticConfig`... They happened in the same commit, but I think I fixed these tests before creating the `discoverer.StaticConfig` type.